frontend/service/Autonolas.ts

```diff
@@ -160,20 +160,31 @@ const getAvailableRewardsForEpoch = async (): Promise<number | undefined> => {
 /**
  * function to get the staking contract info
  */
-const getStakingContractInfo = async (): Promise<
-  StakingContractInfo | undefined
-> => {
+const getStakingContractInfo = async (
+  serviceId: number,
+): Promise<StakingContractInfo | undefined> => {
+  if (!serviceId) return;
+
   const contractCalls = [
     serviceStakingTokenMechUsageContract.availableRewards(),
     serviceStakingTokenMechUsageContract.maxNumServices(),
     serviceStakingTokenMechUsageContract.getServiceIds(),
```

Comment:
Are these ServiceIds and the ones first of which we pass to the getStakingContractInfo with this new change (I mean the `services` from `services?.[0]?.chain_data?.token`) kind of the same? Just wonder, if they are, maybe we can not do unnecessary call?

Reply:
yep I saw that too, was just wondering if the values are the same